aws_lambda_powertools/utilities/feature_toggles/configuration_store.py

```diff
@@ -0,0 +1,207 @@
+                context_value = rules_context.get(restriction.get(RESTRICTION_KEY, ""))
+                if not context_value:
+                    logger.debug(
+                        "rule did not not match key",
+                        rule_name=rule_name,
+                        default_value=rule_default_value,
+                        feature_name=feature_name,
+                    )
+                    is_match = False  # rule doesn't match, continue to next one
+                    break
+                elif not self._match_by_action(
+                    restriction.get(RESTRICTION_ACTION), restriction.get(RESTRICTION_VALUE), context_value
+                ):
+                    logger.debug(
+                        "rule did not match action",
+                        rule_name=rule_name,
+                        default_value=rule_default_value,
+                        feature_name=feature_name,
+                    )
+                    is_match = False  # rules doesn't match restriction
+                    break
+            # if we got here, all restrictions match
+            if is_match:
+                logger.debug(
+                    "rule matched", rule_name=rule_name, default_value=rule_default_value, feature_name=feature_name
+                )
+                return rule_default_value
+        # no rule matched, return default value of feature
+        logger.debug(
+            "no rule matched, returning default value of feature",
+            default_value=default_value,
+            feature_name=feature_name,
+        )
+        return default_value
+
+    def get_configuration(self, logger: object) -> Dict[str, Any]:
+        """Get configuration string from AWs AppConfig and returned the parsed JSON dictionary
+
+        Args:
+            logger (object): logger class to use. must have a debug, info, error, warning and exception functions that
+            receive a message and kwargs
+
+        Raises:
+            ConfigurationException: Any validation error or appconfig error that can occur
+
+        Returns:
+            Dict[str, Any]: parsed JSON dictionary
+        """
+        try:
+            schema = self._conf_store.get(
+                name=self._conf_name,
+                transform=TRANSFORM_TYPE,
+                max_age=self._cache_seconds,
+            )  # parse result conf as JSON, keep in cache for self.max_age seconds
+        except (GetParameterError, TransformParameterError) as exc:
+            error_str = f"unable to get AWS AppConfig configuration file, exception={str(exc)}"
+            logger.error(error_str)
+            raise ConfigurationException(error_str)
+        # validate schema
+        if not self._validate_json_schema(schema):
+            error_str = "AWS AppConfig schema failed validation"
+            logger.error(error_str)
+            raise ConfigurationException(error_str)
+        return schema
+
+    def get_feature_toggle(
```

Comment:
Given the number of positional arguments, and how they may become confusing in the future, wouldn't it be better to use keyword arguments?

Reply:
I'm not sure i'd want to add more param here in order to keep it simple, but sure, will fix.